### PR TITLE
Userrating in MusicOSD

### PR DESCRIPTION
**Pull request by razzeee**
This adds our 1-10 userrating dialogselect as a bindable key. So skins can call "showrating" and it will show up. Should also work for keymaps.

It also changes the onscreen rating display as you change it. 
The only drawback is, that as long as the song you just rated is playing, you can reopen the rating dialog and won't see the rating preselected. If it was rated previous to your play, this will work.

I also need a skinner to change redo my skin work, as mine was just for tests.
So any takers? @phil65 @HitcherUK ?

### CODE REVIEW

**Comment by phil65** [comment]
wouldnt "setrating" be more appropriate?

**Comment by Razzeee** [comment]
Well it's also showsubtitles

**Comment by phil65** [comment]
"showsubtitles" does something entirely different. ;)
i would take increaserating/decreaserating as a reference, in that case the first part of the word indicates what we actually do with the rating.

**Comment by Montellese** [comment]
Since it throws up a dialog to change the rating IMO it should be "setrating" and not "showrating".

**Comment by Razzeee** [comment]
Well I certainly don't care what name we use, if you guys want setrating, let's do it.

**Comment by Razzeee** [comment]
Changed to "setrating"

Also got the skin work from @HitcherUK 
Thanks!

image

**Comment by Tolriq** [comment]
@Razzeee anyway to have also the rating updated when updating song rating for currently playing song with AudioLibrary.SetSongDetails ?

**Comment by Razzeee** [comment]
I'll have to look into that, have you tried it and it doesn't work?

**Comment by Tolriq** [comment]
Not with this patch, but it's like that since I implemented song ratings in Yatse. I don't see anything in this PR that could change that.

**Comment by Razzeee** [comment]
Correct, shouldn't be affected by this PR. But feels weird as it updates fine, when I set it from the new popup window. Will need some investigation.

**Comment by Tolriq** [comment]
Well from your code I suppose the JSON part just miss a call to 
```
CGUIMessage msg(GUI_MSG_NOTIFY_ALL, 0, 0, GUI_MSG_UPDATE_ITEM, 0, track);
``` but it's pure guess.

**Comment by Razzeee** [comment]
Correct, it's due to that call. But I'm not sure if it's okay to call that from the json methods. @Montellese

**Comment by Razzeee** [comment]
jenkins build this please

**Comment by un1versal** [comment]
This broke the OSD controls alignment to left hand side, its now off by 3 miles from where it should be

File: addons/skin.confluence/720p/MusicOSD.xml [hunk 1]
```diff
@@ -258,10 +258,6 @@
 				<visible>!MusicPlayer.Content(LiveTV)</visible>
 				<onclick>PlayerControl(Random)</onclick>
 			</control>
-			<control type="image" id="2300">
```
Comment (un1versal) [inline addons/skin.confluence/720p/MusicOSD.xml]: please restore proper alignment of OSD elements to far left

Comment (un1versal) [inline addons/skin.confluence/720p/MusicOSD.xml]: To restore correct aligment use
```
			<control type="image" id="2300">
				<width>105</width>
				<texture>-</texture>
			</control>
```

Comment (HitcherUK) [inline addons/skin.confluence/720p/MusicOSD.xml]: @Razzeee not sure why I missed that but un1versal's fix corrects it

**Pull request merged**